File: iniciante/python/1018.py
valor = int(input())
valor_inicial = valor

n1, n2, n5, n10, n20, n50, n100 = 0,0,0,0,0,0,0

if valor >= 100:
    n100 = valor // 100
    valor = valor - n100 * 100
if valor >= 50:
    n50 = valor // 50
    valor = valor - n50 * 50
if valor >= 20:
    n20 = valor // 20
    valor = valor - n20 * 20
if valor >= 10:
    n10 = valor // 10
    valor = valor - n10 * 10
if valor >= 5:
    n5 = valor // 5
    valor = valor - n5 * 5
if valor >= 2:
    n2 = valor // 2
    valor = valor - n2 * 2
if valor >= 1:
    n1 = valor // 1
    valor = valor - n1 * 1

print(valor_inicial)
print(f'{n100} nota(s) de R$ 100,00')
print(f'{n50} nota(s) de R$ 50,00')
print(f'{n20} nota(s) de R$ 20,00')
print(f'{n10} nota(s) de R$ 10,00')
print(f'{n5} nota(s) de R$ 5,00')
print(f'{n2} nota(s) de R$ 2,00')
print(f'{n1} nota(s) de R$ 1,00')

File: iniciante/c/1018.c
#include <stdio.h>

int main(){
    int n1 = 0, n2 = 0, n5 = 0, n10 = 0, n20 = 0, n50 = 0, n100 = 0;
    int number, ini;
    scanf("%d", &number);
    ini = number;

    if(number>=100){
        n100 = number / 100;
        number = number - n100 * 100;
    }
    if(number>=50){
        n50 = number / 50;
        number = number - n50 * 50;
    }
    if(number>=20){
        n20 = number / 20;
        number = number - n20 * 20;
    }
    if(number>=10){
        n10 = number / 10;
        number = number - n10 * 10;
    }
    if(number>=5){
        n5 = number / 5;
        number = number - n5 * 5;
    }
    if(number>=2){
        n2 = number / 2;
        number = number - n2 * 2;
    }
    if(number>=1){
        n1 = number / 1;
        number = number - n1 * 1;
    }

    printf("%d\n", ini);
    printf("%d nota(s) de R$ 100,00\n", n100);
    printf("%d nota(s) de R$ 50,00\n", n50);
    printf("%d nota(s) de R$ 20,00\n", n20);
    printf("%d nota(s) de R$ 10,00\n", n10);
    printf("%d nota(s) de R$ 5,00\n", n5);
    printf("%d nota(s) de R$ 2,00\n", n2);
    printf("%d nota(s) de R$ 1,00\n", n1);
    return 0;
}